components/script/script_thread.rs

```diff
@@ -1269,11 +1269,15 @@ impl ScriptThread {
 
         let task_queue = TaskQueue::new(port, chan.clone());
 
-        let background_hang_monitor = state.background_hang_monitor_register.register_component(
-            MonitoredComponentId(state.id, MonitoredComponentType::Script),
-            Duration::from_millis(1000),
-            Duration::from_millis(5000),
-        );
+        let state_id = state.id.clone();
```

Comment:
You can't do this inside the `map`?

Reply:
No, because the value of `state` gets moved in line 1266 (`state.control_port`) and is then borrowed and used inside above closure which throws the following error-
```
error[E0382]: borrow of moved value: `state`
    --> components/script/script_thread.rs:1274:90
     |
1266 |         let control_port = ROUTER.route_ipc_receiver_to_new_crossbeam_receiver(state.control_port);
     |                                                                                ------------------ value moved here
...
1274 |         let background_hang_monitor = state.background_hang_monitor_register.clone().map(|bhm| {
     |                                                                                          ^^^^^ value borrowed here after partial move
1275 |             bhm.register_component(
1276 |                 MonitoredComponentId(state.id.clone(), MonitoredComponentType::Script),
     |                                      ----- borrow occurs due to use in closure
     |
     = note: move occurs because `state.control_port` has type `ipc_channel::ipc::IpcReceiver<script_traits::ConstellationControlMsg>`, which does not implement the `Copy` trait
```

Reply:
Ok. What about if you do `&state.id.clone()`?

Reply:
Nope, error-"expected struct, found reference".

Reply:
Ah Ok, I think it will work with `state.id.clone()` if you move the `let control_port =` to below `let background_hang_monitor =`(which I think it possible)

Reply:
This one worked!